### PR TITLE
feat(settings): Set projen-managed topic for projen managed repositories through settings.yml [CU-8x8uu0gmy]

### PR DESCRIPTION
**Pull request by swyrwiak-cu**
As discussed on https://click-up.slack.com/archives/C01SW5QG7PX/p1681229005165429 we will be using required workflows for repositories that are not managed by **projen**. 
I need a way to identify **projen** managed repositories. I want to do it through topics, that can be set on the repo level and then parsed through `gh api`

### CODE REVIEW

File: test/github-settings.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('addGitHubSettingsYml', () => {
+  test('file added', () => {
+    const project = new typescript.TypeScriptProject({
+      defaultReleaseBranch: 'main',
+      name: 'test',
+    });
+    gitHubSettings.addGitHubSettingsYml(project);
+    const synth = Testing.synth(project);
+    expect(synth['.github/workflows/semgrep.yml']).toMatchSnapshot();
```
Comment (nathanhruby) [inline test/github-settings.test.ts:13]: I don't grok how semgrep gets added here?  shouldn't this be settings.yml?

File: src/github-settings.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { typescript, YamlFile } from 'projen';
+
+export module gitHubSettings {
+  export const REPOSITORY_TOPICS = 'projen-managed';
```
Comment (nathanhruby) [inline src/github-settings.ts:4]: if this is a list, can we make it one now?

File: src/github-settings.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { typescript, YamlFile } from 'projen';
+
+export module gitHubSettings {
```
Comment (ahammond) [inline src/github-settings.ts:3]: Missing docstring. This is library code. Must at the very least reference https://github.com/apps/settings

File: test/github-settings.test.ts (hunk 1 above)
Comment (ahammond) [reply test/github-settings.test.ts:13]: The semgrep stuff should be in a different PR, I think. Maybe is?

**Review by ahammond — CHANGES_REQUESTED** [review]

File: test/github-settings.test.ts (hunk 1 above)
Comment (swyrwiak-cu) [reply test/github-settings.test.ts:13]: There is a separate PR that adds `semgrep` to `projen` and it's https://github.com/time-loop/clickup-projen/pull/147

**Pull request closed without merging**